### PR TITLE
[hostcfgd] Fix Boolean String Evaluation

### PR DESCRIPTION
**Pull request by tahmed-dev**
New attribute 'has_timer' introduced to init_cfg.json does not evaluate
as Bool, rather it evaluates as string. This PR fixes this issue. Also,
this PR fixes an issue when there is system config unit (snmp. telemetry) 
has no installation config (WantedBy=, RequiredBy=, Also=, Alias=) settings
in the [Install] section. In the latter case, the .service should not be enabled.

signed-off-by: Tamer Ahmed <tamer.ahmed@microsoft.com>

The issue appears to be stemming from config_db as the new has_timer is stored as string in the config_db
```
    "FEATURE": {
        "bgp": {
            "auto_restart": "enabled", 
            "has_timer": "False", 
            "high_mem_alert": "disabled", 
            "state": "enabled"
        }, 
        "database": {
            "auto_restart": "disabled", 
            "has_timer": "False", 
            "high_mem_alert": "disabled", 
            "state": "enabled"
        }, 
        "dhcp_relay": {
            "auto_restart": "enabled", 
            "has_timer": "False", 
            "high_mem_alert": "disabled", 
            "state": "enabled"
        }, 
        "lldp": {
            "auto_restart": "enabled", 
            "has_timer": "False", 
            "high_mem_alert": "disabled", 
            "state": "enabled"
        }, 
        "mgmt-framework": {
            "auto_restart": "enabled", 
            "has_timer": "False", 
            "high_mem_alert": "disabled", 
            "state": "enabled"
        }, 
        "nat": {
            "auto_restart": "enabled", 
            "has_timer": "False", 
            "high_mem_alert": "disabled", 
            "state": "disabled"
        }, 
        "pmon": {
            "auto_restart": "enabled", 
            "has_timer": "False", 
            "high_mem_alert": "disabled", 
            "state": "enabled"
        }, 
        "radv": {
            "auto_restart": "enabled", 
            "has_timer": "False", 
            "high_mem_alert": "disabled", 
            "state": "enabled"
        }, 
        "sflow": {
            "auto_restart": "enabled", 
            "has_timer": "False", 
            "high_mem_alert": "disabled", 
            "state": "disabled"
        }, 
        "snmp": {
            "auto_restart": "enabled", 
            "has_timer": "True", 
            "high_mem_alert": "disabled", 
            "state": "enabled"
        }, 
        "swss": {
            "auto_restart": "enabled", 
            "has_timer": "False", 
            "high_mem_alert": "disabled", 
            "state": "enabled"
        }, 
        "syncd": {
            "auto_restart": "enabled", 
            "has_timer": "False", 
            "high_mem_alert": "disabled", 
            "state": "enabled"
        }, 
        "teamd": {
            "auto_restart": "enabled", 
            "has_timer": "False", 
            "high_mem_alert": "disabled", 
            "state": "enabled"
        }, 
        "telemetry": {
            "auto_restart": "enabled", 
            "has_timer": "True", 
            "high_mem_alert": "disabled", 
            "state": "enabled", 
            "status": "enabled"
        }
    }, 
```
**- Why I did it**
To fix service start noticed during test.

**- How I did it**
Add code to properly evaluate newly added field

**- How to verify it**
config load_minigraph does not shoe lldp.timer as started.

**- Which release branch to backport (provide reason below if selected)**
N/A

- [ ] 201811
- [x] 201911
- [x] 202006

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: files/image_config/hostcfgd/hostcfgd [hunk 1]
```diff
@@ -41,15 +42,15 @@ def obfuscate(data):
...
     if state == "enabled":
         start_cmds = []
         for suffix in feature_suffixes:
             start_cmds.append("sudo systemctl unmask {}.{}".format(feature_name, suffix))
-            start_cmds.append("sudo systemctl enable {}.{}".format(feature_name, suffix))
-        # If feature has timer associated with it, start corresponding systemd .timer unit
-        # otherwise, start corresponding systemd .service unit
+        # If feature has timer associated with it, start/enable corresponding systemd .timer unit
+        # otherwise, start/enable corresponding systemd .service unit
+        start_cmds.append("sudo systemctl enable {}.{}".format(feature_name, feature_suffixes[-1]))
```
Comment (jleveque) [inline files/image_config/hostcfgd/hostcfgd:53]: Can you also please add detail of this new change to the PR description?

Comment (tahmed-dev) [reply files/image_config/hostcfgd/hostcfgd:53]: Updated.

**Comment by tahmed-dev** [comment]
retest vsimage please

**Comment by abdosi** [comment]
@tahmed-dev not able to completely follow PR description . 
Eg: snmp.timer has Install section but still we do do not want to enable snmp.service ?
[Install]
WantedBy=timers.target

**Comment by tahmed-dev** [comment]
> @tahmed-dev not able to completely follow PR description .
> Eg: snmp.timer has Install section but still we do do not want to enable snmp.service ?
> [Install]
> WantedBy=timers.target

@abdosi, the issue was reported by systemd as follows:
```
Aug 25 18:47:59.248264 str-s6000-acs-14 INFO hostcfgd[22667]: The unit files have no installation config (WantedBy=, RequiredBy=, Also=,
Aug 25 18:47:59.250047 str-s6000-acs-14 INFO hostcfgd[22667]: Alias= settings in the [Install] section, and DefaultInstance= for template
Aug 25 18:47:59.251773 str-s6000-acs-14 INFO hostcfgd[22667]: units). This means they are not meant to be enabled using systemctl.
Aug 25 18:47:59.254822 str-s6000-acs-14 INFO hostcfgd[22667]:  
Aug 25 18:47:59.256584 str-s6000-acs-14 INFO hostcfgd[22667]: Possible reasons for having this kind of units are:
Aug 25 18:47:59.262071 str-s6000-acs-14 INFO hostcfgd[22667]: • A unit may be statically enabled by being symlinked from another unit's
Aug 25 18:47:59.263599 str-s6000-acs-14 INFO hostcfgd[22667]:   .wants/ or .requires/ directory.
Aug 25 18:47:59.265389 str-s6000-acs-14 INFO hostcfgd[22667]: • A unit's purpose may be to act as a helper for some other unit which has
Aug 25 18:47:59.268989 str-s6000-acs-14 INFO hostcfgd[22667]:   a requirement dependency on it.
Aug 25 18:47:59.270647 str-s6000-acs-14 INFO hostcfgd[22667]: • A unit may be started when needed via activation (socket, path, timer,
Aug 25 18:47:59.272414 str-s6000-acs-14 INFO hostcfgd[22667]:   D-Bus, udev, scripted systemctl call, ...).
Aug 25 18:47:59.274089 str-s6000-acs-14 INFO hostcfgd[22667]: • In case of template units, the unit is meant to be enabled with some
Aug 25 18:47:59.275855 str-s6000-acs-14 INFO hostcfgd[22667]:   instance name specified.
Aug 25 18:47:59.279172 str-s6000-acs-14 INFO hostcfgd: Running cmd: 'sudo systemctl start snmp.service'
```

**Comment by tahmed-dev** [comment]
retest vsimage please

**Comment by tahmed-dev** [comment]
retest broadcom please

**Comment by tahmed-dev** [comment]
retest broadcom please

**Comment by jleveque** [comment]
Retest vsimage please

**Comment by tahmed-dev** [comment]
retest vsimage please

**Pull request merged**